powerman@dr.drummie: Add new applet

drdrummie opened the pull request:
powerman@dr.drummie: Add new applet

Upgraded version of the original Cinnamon power applet with added automation. For laptops, it automatically switches brightness and power profiles when changing between AC and battery, includes low‑battery protection, customizable settings, and offers an option to replace the default power applet or hide this one.
While partially based on existing code, it introduces additional functionality and customization features.

All usage instructions are provided in the README.md.

drdrummie committed:
Create info.json

drdrummie committed:
Added applet's files

drdrummie committed:
Refactor README.md: Update features section for clarity and remove outdated technical details

drdrummie committed:
Fix: Update original power applets tracking and metadata details

drdrummie committed:
Fix: Update default brightness values for AC and idle settings

drdrummie committed:
Fix: Update applet icon attribution in README.md and replace icon image

drdrummie committed:
Fix: Update README.md for clarity and consistency in feature descriptions

drdrummie committed:
Update signal connection method for brightness change

Should fix slider issues..

drdrummie committed:
Update README with brightness control details issues

drdrummie committed:
New version of the applet 1.9.1.
- Refactored whole code structure, separated automation logic into managers, fixed bugs
- Improved speed and responsiveness of the applet
- Updated README.md
- Added CHANGELOG.md
- Updated translations pot and po files

drdrummie committed:
v1.9.1 Fix:
- Improve power state detection for full batteries and AC connection

drdrummie committed:
New version of the applet 1.9.5.
- Extended default popup with switches for quickly enable / disable automation.
- Added option to remember user's "external" change of brightness with settings depending on power source.
- Updated po / pot.

drdrummie commented:
@fredcw 
I believe they can run at the same time, you should be able to hide powerman if don't want it into system tray. 
I saw some battery applets but didn't see the option to automatically switch power mode or brightness. 
I've tried my own implementation, but power applet seemed as the right way to go.. Also didn't found that automation option in Mint so tried to made it with this applet.. 
I've tested it on 3 different laptops, haven't see the issues, still - my own are older ones so I can't really tell how it behaves on modern systems..

drdrummie committed:
New version of the applet 1.9.6.
- Fixed signal leaks and stability fixes
- Updated README and CHANGELOG

claudiux commented:
It is possible to notify the users that the `power@cinnamon.org` applet is also in use and give them the option to disable it. See the code from `sound150@claudiux`, which detects `sound@cinnamon.org`.

drdrummie commented:
Sure, no problem, I'll look into it..
Applet already handles removal of original power applet, so I need to see about notification..

drdrummie committed:
New version of the applet 1.9.10.

- Fixed AC source detection for laptops with charge-protection circuits
- Fixed brightness D-Bus race conditions and CSD compatibility
- Fixed battery saver threshold detection
- Replaced "Replace system applet" switch with popup "Remove power applet" button
- Added missing translations for 17 languages
- Updated CHANGELOG
